airflow/providers/amazon/aws/transfers/google_api_to_s3.py

```diff
@@ -167,21 +167,21 @@ def _retrieve_data_from_google_api(self):
         )
         return google_api_response
 
-    def _load_data_to_s3(self, data):
+    def _load_data_to_s3(self, data: dict) -> None:
         s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
         s3_hook.load_string(
             string_data=json.dumps(data), key=self.s3_destination_key, replace=self.s3_overwrite
         )
 
-    def _update_google_api_endpoint_params_via_xcom(self, task_instance):
+    def _update_google_api_endpoint_params_via_xcom(self, task_instance: TaskInstance) -> None:
         google_api_endpoint_params = task_instance.xcom_pull(
             task_ids=self.google_api_endpoint_params_via_xcom_task_ids,
             key=self.google_api_endpoint_params_via_xcom,
         )
         self.google_api_endpoint_params.update(google_api_endpoint_params)
 
-    def _expose_google_api_response_via_xcom(self, task_instance, data):
+    def _expose_google_api_response_via_xcom(self, task_instance: TaskInstance, key: str, data: dict) -> None:
         if sys.getsizeof(data) < MAX_XCOM_SIZE:
-            task_instance.xcom_push(key=self.google_api_response_via_xcom, value=data)
+            task_instance.xcom_push(key=key, value=data)
```

Comment:
The argument `key` of `xcom_push()` method requires `str` value but `self.google_api_response_via_xcom` is `Optional[str]` so we need to check that it's not None. In the outer scope of this method it's checked already, so I change the signature of this method to receive it from the outer method.

Reply:
How about adding
```
key = self.google_api_response_via_xcom or XCOM_RETURN_KEY
```
between L151 and L152?

Reply:
How about doing:
```
key = self.google_api_response_via_xcom or XCOM_RETURN_KEY
```
instead of changing the signature?

Reply:
Thank you. Using XCOM_RETURN_KEY is simpler than changing the signature. I'll fix it.